packages/world/src/modules/keysintable/KeysInTableDynamicFieldIndex.sol

```diff
@@ -0,0 +1,49 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.0;
+
+import { StoreSwitch } from "@latticexyz/store/src/StoreSwitch.sol";
+import { Bytes } from "@latticexyz/store/src/Bytes.sol";
+
+import { KeysInTable, KeysInTableTableId } from "./tables/KeysInTable.sol";
+
+/**
+ * Based on `KeysInTable`, but allows dynamically passing fieldIndex.
+ * Dynamic fieldIndex relies on all fields having the same schema type.
```

Comment:
Can you elaborate on this comment a bit? Do you mean the field types of `KeysInTable`? If so, since they all have the same type (bytes32 array since it's all keys), do we need this comment?

Reply:
yeah, the point is this trick shouldn't be used for other tables unless all their fields have the same type. I just didn't think it was obvious what's going on there so explicitly pointed it out